### PR TITLE
Entity management (by admin). Fields configuration. Field as a dropdown.

### PR DESCRIPTION
**Pull request by GalinaRuban**
Entity management (by admin). Fields configuration (Existing record). Advanced. Field as a dropdown.

https://trello.com/c/YcGISgCa/736-entity-management-by-admin-fields-configuration-existing-record-advanced-field-as-a-dropdown

### CODE REVIEW

**Review by guram21 — APPROVED** [review]

File: src/test/java/utils/TestUtils.java [hunk 1]
```diff
@@ -27,7 +27,7 @@ public static void click(WebDriver driver, WebElement element) {
 
     public static void scroll(WebDriver driver, WebElement element) {
         JavascriptExecutor executor = (JavascriptExecutor) driver;
-        executor.executeScript("arguments[0].scrollIntoView();", element);
+        executor.executeScript("arguments[0].scrollIntoView(true);", element);
```
Comment (oooleks) [inline src/test/java/utils/TestUtils.java:30]: The default value of alignToTop argument of scrollIntoView() method is **true**, hence no need to mention it, but **false** only when needed.

File: src/test/java/model/admin/edit/EntityEditPage.java [hunk 2]
```diff
@@ -104,4 +104,17 @@ public EntityTablePage clickCreateNewChildFolder() {
...
-}
\ No newline at end of file
+
+    public List<String> getDropdownValues(String fieldName) {
+        List<WebElement> dropdownLabelsDisplayed = getDriver().findElements(By.xpath(String.format("//select[@id='%s']/option", fieldName)));
+        List<String> dropdownValues = new ArrayList<>();
+        for (int i = 0; i < dropdownLabelsDisplayed.size(); i++) {
+            dropdownValues.add(dropdownLabelsDisplayed.get(i).getText());
+        }
+        return dropdownValues;
```
Comment (oooleks) [inline src/test/java/model/admin/edit/EntityEditPage.java:114]: You can use collector and rewrite it all like:
return getDriver().findElements(By.xpath(String.format("//select[@id='%s']/option", fieldName))).stream().map(WebElement::getText).collect(Collectors.toList());

File: src/test/java/model/admin/configuration/entities/AdminFieldsPage.java [hunk 3]
```diff
@@ -176,6 +177,14 @@ public AdminFieldsConfigPage clickConfigWheelButton() {
         return new AdminFieldsConfigPage(getDriver());
     }
 
+    public AdminFieldsConfigPage clickConfigWheelButton(String fieldName) {
+        WebElement configBtnOfSpecificField = getDriver().findElement(By.xpath(String.format("//tr/td[.=contains(text(), '%s')]/following-sibling::td/a//i[.='settings']", fieldName)));
+        ((JavascriptExecutor) getDriver()).executeScript("arguments[0].scrollIntoView(true);", configBtnOfSpecificField);
```
Comment (oooleks) [inline src/test/java/model/admin/configuration/entities/AdminFieldsPage.java:182]: Use TestUtils.scroll method instead which is exactly the same

File: src/test/java/model/admin/configuration/entities/AdminFieldsConfigPage.java [hunk 4]
```diff
@@ -60,8 +61,8 @@ public AdminFieldsConfigPage setPlaceholderValues(String updatePlaceholder) {
     }
 
     public AdminFieldsPage clickSaveButton() {
-        TestUtils.scroll(getDriver(), saveButton);
-        TestUtils.click(getWait(), saveButton);
+        ((JavascriptExecutor) getDriver()).executeScript("arguments[0].scrollIntoView(true);", saveButton);
```
Comment (oooleks) [inline src/test/java/model/admin/configuration/entities/AdminFieldsConfigPage.java:64]: Use TestUtils.scroll method

File: src/test/java/model/admin/edit/EntityEditPage.java (hunk 2 above)
Comment (GalinaRuban) [reply src/test/java/model/admin/edit/EntityEditPage.java:114]: ok

File: src/test/java/utils/TestUtils.java (hunk 1 above)
Comment (GalinaRuban) [reply src/test/java/utils/TestUtils.java:30]: ok

File: src/test/java/model/admin/configuration/entities/AdminFieldsPage.java (hunk 3 above)
Comment (GalinaRuban) [reply src/test/java/model/admin/configuration/entities/AdminFieldsPage.java:182]: ok

File: src/test/java/model/admin/configuration/entities/AdminFieldsConfigPage.java (hunk 4 above)
Comment (GalinaRuban) [reply src/test/java/model/admin/configuration/entities/AdminFieldsConfigPage.java]: ok

**Review by oooleks — APPROVED** [review]

**Pull request merged**